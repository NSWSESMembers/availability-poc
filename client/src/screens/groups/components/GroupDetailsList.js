import React from 'react';
import PropTypes from 'prop-types';
import {
  Text,
  FlatList,
  Alert,
} from 'react-native';

import { Center, Container } from '../../../components/Container';

import EventListItem from '../../../components/Events/EventListItem';
import ScheduleListItem from '../../../components/Schedules/ScheduleListItem';
import UserListItem from '../../../components/Users/UserListItem';


const groupDetailsRenderItem = ({ item }) => {
  // eslint-disable-next-line no-underscore-dangle
  switch (item.__typename) { // GQL typename from Schema
    case 'User':
      return (
        <UserListItem
          user={item}
      // TODO: Deep link
          onPress={() => Alert.alert('PLACEHOLDER', 'Deeplink to user details')}
        />
      );
    case 'Schedule':
      return (
        <ScheduleListItem
          schedule={item}
    // TODO: Deep link
          onPress={() => Alert.alert('PLACEHOLDER', 'Deeplink to schedule details')}
        />
      );
    default:
      return (
        <EventListItem
          event={item}
    // TODO: Deep link
          onPress={() => Alert.alert('PLACEHOLDER', 'Deeplink to event details')}
        />
      );
  }
};

groupDetailsRenderItem.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  item: PropTypes.object.isRequired,
};

const GroupDetailsList = ({ items, selectedIndex, networkStatus, refetch }) => (
  <Container>
    <FlatList
      data={items[selectedIndex]}
      ListHeaderComponent={() =>
        (!items[selectedIndex].length ? (
          <Center>
            <Text>Nothing here</Text>
          </Center>
        ) : null)}
      keyExtractor={item => item.id}
      renderItem={groupDetailsRenderItem}
      refreshing={networkStatus === 4}
      onRefresh={refetch}
    />
  </Container>
);

GroupDetailsList.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  items: PropTypes.array.isRequired,
  selectedIndex: PropTypes.number.isRequired,
  networkStatus: PropTypes.number.isRequired,
  refetch: PropTypes.func,
};

export default GroupDetailsList;
