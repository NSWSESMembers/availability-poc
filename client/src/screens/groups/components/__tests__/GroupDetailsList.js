import React from 'react';
import renderer from 'react-test-renderer';

import GroupDetailsList from '../GroupDetailsList';


const groupUserItems = [{
  displayName: 'displayName',
  id: 1,
  username: 'username',
  __typename: 'User',
}];

const groupScheduleItems = [{
  details: 'details',
  endTime: 1,
  id: 2,
  name: 'name',
  startTime: 0,
  __typename: 'Schedule',
}];

const groupEventItems = [{
  details: 'details',
  id: 1,
  name: 'name',
  __typename: 'Event',
}];

test('renders correctly', () => {
  const tree = renderer.create(
    <GroupDetailsList
      items={[groupUserItems, groupScheduleItems, groupEventItems]}
      networkStatus={4}
      selectedIndex={1}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();

  const tree2 = renderer.create(
    <GroupDetailsList
      items={[groupUserItems, groupScheduleItems, groupEventItems]}
      networkStatus={4}
      selectedIndex={2}
    />,
  ).toJSON();
  expect(tree2).toMatchSnapshot();

  const tree3 = renderer.create(
    <GroupDetailsList
      items={[groupUserItems, groupScheduleItems, groupEventItems]}
      networkStatus={4}
      selectedIndex={3}
    />,
  ).toJSON();
  expect(tree3).toMatchSnapshot();
});
