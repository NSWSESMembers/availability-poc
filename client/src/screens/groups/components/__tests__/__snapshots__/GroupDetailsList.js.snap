// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#EEE",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-start",
      },
    ]
  }
>
  <RCTScrollView
    ListHeaderComponent={[Function]}
    data={
      Array [
        Object {
          "__typename": "Schedule",
          "details": "details",
          "endTime": 1,
          "id": 2,
          "name": "name",
          "startTime": 0,
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={undefined}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshing={true}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      />
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 2,
                  "elevation": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 0,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 6,
                  "paddingBottom": 10,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 10,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Object {
                    "flex": 0,
                    "fontWeight": "bold",
                    "includeFontPadding": false,
                    "marginRight": 8,
                    "textAlignVertical": "top",
                  }
                }
              >
                name
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={2}
                style={
                  Object {
                    "color": "#6B6B6B",
                    "flex": 0,
                    "fontSize": 12,
                    "includeFontPadding": false,
                    "textAlignVertical": "bottom",
                  }
                }
              >
                Ongoing
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingLeft": 8,
                  "width": 40,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 30,
                    },
                    Object {
                      "alignSelf": "center",
                    },
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                ÔÅ≥
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`renders correctly 2`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#EEE",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-start",
      },
    ]
  }
>
  <RCTScrollView
    ListHeaderComponent={[Function]}
    data={
      Array [
        Object {
          "__typename": "Event",
          "details": "details",
          "id": 1,
          "name": "name",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={undefined}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshing={true}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      />
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 2,
                  "elevation": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 0,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 6,
                  "paddingBottom": 10,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 10,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Object {
                    "flex": 0,
                    "fontWeight": "bold",
                    "includeFontPadding": false,
                    "marginRight": 8,
                    "textAlignVertical": "top",
                  }
                }
              >
                name
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={2}
                style={
                  Object {
                    "color": "#6B6B6B",
                    "flex": 0,
                    "fontSize": 12,
                    "includeFontPadding": false,
                    "textAlignVertical": "bottom",
                  }
                }
              >
                details
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingLeft": 8,
                  "width": 40,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 30,
                    },
                    Object {
                      "alignSelf": "center",
                    },
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                ÔÇ°
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`renders correctly 3`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#EEE",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-start",
      },
    ]
  }
>
  <RCTScrollView
    ListHeaderComponent={[Function]}
    data={undefined}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={undefined}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshing={true}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      />
    </View>
  </RCTScrollView>
</View>
`;
